gulpfile.js

```diff
@@ -0,0 +1,239 @@
+}
+
+function build(done) {
+    switch(true) {
+        case ( !params.conditions.isRunningOnADO ): 
+            console.log('Running Local Build');
+            gulp.series(lernaBuild)(done);
+        break;
+
+        case ( params.conditions.isRunningOnADO && 
+                !params.build.pullRequestID && 
+                !!params.build.buildSourceBranch.replace(/refs\/heads\/(feature\/)?/i, '').match(/master/ig)):
+            console.log('Running Azure DevOps Release Build');
+            gulp.series( printVersion, adoPrep, toolInstall, lernaBuild, lernaPublish, systemPublish, gitCommit )(done);
+        break;
+
+        case ( params.conditions.isRunningOnADO && 
+                !!params.build.pullRequestID ):
+            console.log('Running Azure DevOps Pull Request Build');
+            gulp.series( printVersion, adoPrep, toolInstall, lernaBuild )(done);
+        break;
+        
+        case ( params.conditions.isRunningOnADO && 
+            params.build.buildReason.match(/manual/ig) ):
+            console.log('Running Azure DevOps Manual Build');
+            gulp.series( printVersion, adoPrep, toolInstall, lernaBuild )(done);
+        break;
+
+        default:
+            done('Build did not complete!');
+        break;        
+    }
+}
+
+function cleanCoverage() {
+    return del('coverage/**', { force: true });
+}
+
+
+function gitCommit(done) {
+    var branchName = params.build.buildSourceBranch;
+    if (branchName !== 'master') {
+        // all branches have refs/heads/ - we don't need that
+        // we will also remove feature/ if it's there
+        branchName = branchName.replace(/refs\/heads\/(feature\/)?/i, '');
+    }    
+    var gitScript = `sudo git checkout ${branchName} && 
+    sudo git config user.email '${params.build.buildRequestedForEmail}' &&
+    sudo git config user.name '${params.build.buildRequestedFor}' &&
+    sudo git add . && 
+    sudo git commit --author '${params.build.buildRequestedFor} <${params.build.buildRequestedForEmail}>' --message "[skip ci][CHORE] Update & Publish" && 
+    sudo git tag v ${lerna.version} &&
+    sudo git push origin ${branchName} &&
+    sudo git push origin --tags`;
+    console.log('Git Script: ' + gitScript);
+    return shell.task(gitScript)(done());
+}
+
+function toolInstall(done) {
+    var gitScript = `sudo npm install lerna@3.13.0 typescript@3.3.3 --global` ;
+    console.log('Tool Script: ' + gitScript);
+    return runCmd(gitScript, done);    
+}
+
+function lernaBuild(done) {
+    var gitScript = `sudo npm run clean:build` ;
+    console.log('Build Script: ' + gitScript);
+    return runCmd(gitScript, done);    
+}
+
+function lernaPublish(done) {    
+    var gitScript = `sudo npm run publish` ;
+    console.log('Build Script: ' + gitScript);
+    return runCmd(gitScript, done);    
+}
+
+function systemPublish(done) {
+    var gitScript = `sudo npm publish --prefix ./system &&  npm run release-build --prefix ./system` ;
```

Comment:
think we only want release-build here, lerna publish above takes care of all the npm publishing